common/src/chunk_collision.rs

```diff
@@ -288,40 +313,42 @@ fn solve_sphere_plane_intersection(
 /// intersects the given line.
 fn solve_sphere_line_intersection(
     ray: &Ray,
-    line_position: &na::Vector4<f32>,
-    line_direction: &na::Vector4<f32>,
-    cosh_radius: f32,
+    line_normal0: &na::Vector4<f32>,
+    line_normal1: &na::Vector4<f32>,
+    sinh_radius: f32,
 ) -> Option<f32> {
-    // This could be made more numerically stable by using a formula that depends on sinh_radius,
-    // but the precision requirements of collision should be pretty lax.
```

Comment:
Why is sinh more precise? Can we capture the required precision in a test that the cosh version fails?

Reply:
Might be a good idea. I'll see if I can write a test for this.

To answer your question, `cosh(x)` is approximately `1 + x^2`, while `sinh(x)` is approximately `x`. When `x` is small, then `cosh(x)` is very close to 1, and solving for `x` geometrically is like trying to find the intersection point of two almost-parallel lines.

Reply:
I've added a test to demonstrate the difference for edges (and I believe it's safe to extrapolate this reasoning for vertices). The test is a bit lengthy and convoluted, so I'm definitely open to feedback on it.

Reply:
LGTM. I wouldn't mind trimming out the parts that demonstrate the less precise method; my main goal is just having some insurance against future regression.

Reply:
That makes sense to me. I definitely agree with you here, so I'll go ahead and get rid of the other checks.